### PR TITLE
Backend/crud venue - this is for updating the reviews 

### PR DESCRIPTION
**Pull request by kaamil2**
# Description

[Link to Ticket](insert the link to your ticket inside the parenthesis here)

Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. If they are unit
tests, provide the file name the tests are in. If they are not unit tests,
describe how you tested the change.
Screenshot 2024-09-28 at 4 05 04 PM
Screenshot 2024-09-28 at 4 05 09 PM

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have reached out to another developer to review my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/internal/storage/storage.go [hunk 1]
```diff
@@ -4,13 +4,30 @@ import (
...
+	GetAllVenues(context.Context) ([]models.Test, error)
+}
+
+type Review interface {
+	PatchVenueReview(ctx context.Context,overall_rating int8,ambiance_rating int8,music_rating int8, crowd_rating int8,service_rating int8,review_text string, venueID uuid.UUID, reviewID int8) error
+}
+
+type GetReviewByID interface {
+	GetVenueReviews(ctx context.Context, review_id int8, venue_id uuid.UUID) ([]models.Review, error)
+}
```
Comment (CamPlume1) [inline backend/internal/storage/storage.go:33]: Please put this function under the "reviews" interface

File: backend/internal/db/db.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package db
```
Comment (CamPlume1) [inline backend/internal/db/db.go:1]: What's up with this? Believe we already have a database connection

File: backend/bin/nightlife
Comment (CamPlume1) [inline backend/bin/nightlife:1]: Can you please remove the binary build as well

**Review by CamPlume1 — APPROVED** [review]
Looks Great

File: package-lock.json
Comment (garrettladley) [inline package-lock.json:1]: untrack?

File: backend/internal/storage/postgres/venues.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+func (db *DB) GetVenueReviews(ctx context.Context, reviewID int8, venueID uuid.UUID) ([]models.Review, error) {
+	// Use proper SQL syntax without single quotes around the table name
+	query := `SELECT * FROM "Review" r WHERE review_id = $1 AND venue_id = $2`
+	rows, err := db.conn.Query(ctx, query, reviewID, venueID)
+	if err != nil {
+		return nil, err // Return nil slice on error for idiomatic Go
+	}
+	defer rows.Close() // Ensure you close the rows to free resources
+
+	var reviews []models.Review
```
Comment (garrettladley) [inline backend/internal/storage/postgres/venues.go:29]: check out [`pgx.RowToStructByName`](https://pkg.go.dev/github.com/jackc/pgx/v5#RowToStructByName)

File: backend/internal/handlers/venue/venue.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+// PatchVenueReview handles the PATCH request to update a review
+func (s *Service) PatchVenueReview(c *fiber.Ctx) error {
+	// Retrieve and validate the venueID and reviewID from the URL parameters
+	venueString := c.Params("venueId")
+
+	venueID, err := uuid.Parse(venueString)
+	if err != nil {
+		return c.Status(http.StatusBadRequest).JSON(fiber.Map{
```
Comment (garrettladley) [inline backend/internal/handlers/venue/venue.go:19]: reuse the err helpers from the `errs` package

File: backend/internal/handlers/venue/routes.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package venue
+
+//NOTE: This is an example usage for auth demonstration purposes. In real configurations (beyond login) all route groups should be protected
```
Comment (garrettladley) [inline backend/internal/handlers/venue/routes.go:3]: remove?

**Review by garrettladley — COMMENTED** [review]
i'd clean up the debug prints then looks good to merge!

**Pull request closed without merging**